user_guide_src/source/database/query_builder/118.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+$data = [
+    [
+        'order'   => 48372,
+        'line'    => 3,
+        'product' => 'Keyboard',
+        'qty'     => 1,
+    ],
+    [
+        'order'   => 48372,
+        'line'    => 4,
+        'product' => 'Mouse',
+        'qty'     => 1,
+    ],
+    [
+        'order'   => 48372,
+        'line'    => 5,
+        'product' => 'Monitor',
+        'qty'     => 2,
+    ],
+];
+
+$builder->setAlias('del')->where('order_line.qty >', 1)->deleteBatch($data, 'order, line');
```

Comment:
I don't know what `setAlias()` means when I see this line.

Reply:
`setDataAlias()` is a better name? 
Or do we really need this method?

> This allows setting an alias for the psuedo table generated with ``setData()``.
https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/database/query_builder.rst#class-reference

Reply:
@codeigniter4/database-team Any comments would be appreciated.

Reply:
I found `setAlias()` is not documented in the changelog:
https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/changelogs/v4.3.0.rst#query-builder

Reply:
The diff between develop and 4.3 BaseBuilder.php public methods:
```diff
--- qb-develop.txt	2022-11-04 16:36:01.000000000 +0900
+++ qb-4.3.txt	2022-11-04 16:36:13.000000000 +0900
@@ -15,6 +15,7 @@
     public function getCompiledQBWhere()
     public function getCompiledSelect(bool $reset = true): string
     public function getCompiledUpdate(bool $reset = true)
+    public function getCompiledUpsert()
     public function getSetData(bool $clean = false): array
     public function getTable(): string
     public function getWhere($where = null, ?int $limit = null, ?int $offset = 0, bool $reset = true)
@@ -30,7 +31,7 @@
     public function ignore(bool $ignore = true)
     public function increment(string $column, int $value = 1)
     public function insert($set = null, ?bool $escape = null)
-    public function insertBatch(?array $set = null, ?bool $escape = null, int $batchSize = 100)
+    public function insertBatch($set = null, ?bool $escape = null, int $batchSize = 100)
     public function join(string $table, $cond, string $type = '', ?bool $escape = null)
     public function like($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
     public function limit(?int $value = null, ?int $offset = 0)
@@ -39,6 +40,7 @@
     public function notHavingLike($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
     public function notLike($field, string $match = '', string $side = 'both', ?bool $escape = null, bool $insensitiveSearch = false)
     public function offset(int $offset)
+    public function onConstraint($set)
     public function orGroupStart()
     public function orHaving($key, $value = null, ?bool $escape = null)
     public function orHavingGroupStart()
@@ -64,6 +66,8 @@
     public function selectSubquery(BaseBuilder $subquery, string $as): self
     public function selectSum(string $select = '', string $alias = '')
     public function set($key, $value = '', ?bool $escape = null)
+    public function setAlias(string $alias): BaseBuilder
+    public function setData($set, ?bool $escape = null, string $alias = '')
     public function setInsertBatch($key, string $value = '', ?bool $escape = null)
     public function setUpdateBatch($key, string $index = '', ?bool $escape = null)
     public function testMode(bool $mode = true)
@@ -71,7 +75,10 @@
     public function union($union)
     public function unionAll($union)
     public function update($set = null, $where = null, ?int $limit = null): bool
-    public function updateBatch(?array $set = null, ?string $index = null, int $batchSize = 100)
+    public function updateBatch($set = null, $constraints = null, int $batchSize = 100)
+    public function updateFields($set, bool $addToDefault = false, ?array $ignore = null)
+    public function upsert($set = null, ?bool $escape = null)
+    public function upsertBatch($set = null, ?bool $escape = null, int $batchSize = 100)
     public function where($key, $value = null, ?bool $escape = null)
     public function whereIn(?string $key = null, $values = null, ?bool $escape = null)
     public function whereNotIn(?string $key = null, $values = null, ?bool $escape = null)
```

Reply:
I'm starting to be confused about some new methods. Do we need `setAlias()` at all? If it's used only by one method, we should add an alias as a second parameter, just like `fromSubquery()` does.

Reply:
`setAlias()` is used by any of the `*Batch()` methods. When calling `setData()` you can pass the alias directly into the method. When setting the data through the `*Batch()` methods there is no parameter for alias. This is where the `setAlias()` method is most useful. 

```php
$db->table('table')
->onConstraint('id')
->setAlias('alias')
->updateFields(['table.name' => 'alias.firstname'])
->upsertBatch($data);
```

>I don't know what setAlias() means, why alias is necessary, when I see this line.

I changed the example to make it more clear. When you use where() and want to reference the data as a pseudo table you need the alias. Also, if you don't set an alias the code will use a default alias name. This could potentially be a problem so you might want to set your own.

Eventually I want to add additional joins and where statements to the `*Batch()` methods. Knowing the alias will be essential for this.

Reply:
IMO, the `setAlias()` method should not be public. Using it alone is completely non-intuitive to me. We have the option to set an alias via `setData()` if we need it.

We should try to maintain some sort of consistency in this area.

Reply:
The generic name `setAlias()` is not a good naming for `*Batch()` method only.

> When calling setData() you can pass the alias directly into the method. When setting the data through the *Batch() methods there is no parameter for alias. This is where the setAlias() method is most useful.

Why don't you call `setData()` if you need an alias?

Reply:
v4.3.0 will come soon.
`setAlias()` is under discussion but has already been included in the `4.3` branch.
So I have created a PR to disable it if it is not ready for the v4.3.0 release. #6871

Reply:
I removed to use of setAlias()